protonvpn_linux_gui/presenters/dashboard_presenter.py

```diff
@@ -0,0 +1,565 @@
+            
+            display_secure_core = get_gui_config("connections", "display_secure_core")
+            secure_core_switch = objects_dict["secure_core"]["secure_core_switch"]
+            secure_core_label_style = objects_dict["secure_core"]["secure_core_label_style"].get_style_context() 
+
+            if display_secure_core == "True":
+                secure_core_switch.set_state(True)
+                secure_core_label_style.remove_class("disabled_label")
+            else:
+                secure_core_switch.set_state(False)
+
+            self.update_labels_server_list(objects_dict, conn_info=conn)
+            self.queue.put(dict(action="hide_dialog"))
+            return_val = True
+        else:
+            self.queue.put(dict(action="update_dialog", label=display_message, spinner=False))
+
+        gui_logger.debug(">>> Ended tasks in \"load_on_start\" thread.")  
+        
+        # return return_val
+
+    def reload_secure_core_servers(self, **kwargs):
+        """Function that reloads server list to either secure-core or non-secure-core.
+        """  
+        # Sleep is needed because it takes a second to update the information,
+        # which makes the button "lag". Temporary solution.
+        time.sleep(1)
+        gui_logger.debug(">>> Running \"update_reload_secure_core_serverslabels_server_list\".")
+
+        return_val = False
+        display_message = "Unable to reload servers!"
+
+        if self.dashboard_service.set_display_secure_core(kwargs.get("update_to")):
+            populate_servers_dict = {
+                "tree_object": kwargs.get("tree_object"),
+                "servers": False
+            }
+            return_val = True
+
+            gobject.idle_add(self.populate_server_list, populate_servers_dict)
+            display_message = "Displaying <b>{}</b> servers!".format("secure-core" if kwargs.get("update_to") == "True" else "non secure-core")
+
+        self.queue.put(dict(action="update_dialog", label=display_message))
+
+        gui_logger.debug(">>> Ended tasks in \"reload_secure_core_servers\" thread.")
+
+        # return return_val
+
+    def connect_to_selected_server(self, **kwargs):
+        """Function that either connects by selected server or selected country.
+        """     
+        user_selected_server = kwargs.get("user_selected_server")
+
+        gui_logger.debug(">>> Running \"openvpn_connect\".")
+            
+        # Check if it should connect to country or server
+        if "#" in user_selected_server:
+            result = self.dashboard_service.connect_to_server(user_selected_server)
+            gui_logger.debug(">>> Log during connection to specific server: {}".format(result))
+        else:
+            result = self.dashboard_service.connect_to_country(user_selected_server)
+            gui_logger.debug(">>> Log during connection to country: {}".format(result))
+
+        display_message = result.stdout.decode()
```

Comment:
If we wrap the CLI better (hence you no longer call it through CLI but rather via native Python) all these stdout.decode() won't be valid anymore. Hence it's better to hide them properly in `connect_to_country`/`connect_to_server` calls.